src/main/java/core/basesyntax/strategy/StrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.service.operation.AddOperationHandlerImpl;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperationHandlerImpl;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StrategyImpl implements Strategy {
+    private final Map<String, OperationHandler> operationHandlerMap;
+
+    {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put("r", new AddOperationHandlerImpl(new StorageDaoImpl()));
```

Comment:
let's create new StorageDaoImpl() in main and pass it to StrategyImpl constructor